planner/core/point_get_plan.go

```diff
@@ -602,6 +611,19 @@ func tryPointGetPlan(ctx sessionctx.Context, selStmt *ast.SelectStmt) *PointGetP
 		p.Handle = intDatum.GetInt64()
 		p.UnsignedHandle = mysql.HasUnsignedFlag(fieldType.Flag)
 		p.HandleParam = handlePair.param
+		if pi != nil {
+			partitionSchema, colNames := buildSchemaForPartition(tblName.Schema, tbl, tblAlias)
+			exprs, err := expression.ParseSimpleExprsWithNames(ctx, pi.Expr, partitionSchema, colNames)
```

Comment:
We can simply check if `pi.Expr` is `*ast.ColumnNameExpr`.

Reply:
Partition express like `column1 + column2 - column3` is also available, not only `ColumnNameExpr`

Reply:
Then we don't use point get for it.

Reply:
if columns in partition express are subset of primary key, we need to use point get.
here is an example:
```
create table t (id int, k int, name varchar(30), primary key(id, k)) partition by hash (id + k) parititions 10;
select name from t where id = 7 and k = 5;
```

Reply:
A hint: of course we can cover more cases with complex code.
When the complex code involves more calculation and covers only some rare cases, it's better to find a balance.